unit_tests/algorithms/UnitTestAlgorithmUtils.h

```diff
@@ -0,0 +1,92 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 National Renewable Energy Laboratory.                  */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+#ifndef UNITTESTALGORITHMUTILS_H
+#define UNITTESTALGORITHMUTILS_H
+
+#include <gtest/gtest.h>
+#include "UnitTestUtils.h"
+
+#include "Algorithm.h"
+#include "SupplementalAlgorithm.h"
+
+namespace unit_test_algorithm_utils {
+
+/** Driver class that mimics Assemble*SolverAlgorithm
+ *
+ * It is the caller's responsibility to populate `activeSuppAlgs_` and call the
+ * `setup` method on the activated SuppAlg before calling the `execute` method
+ * of this class.
+ *
+ */
+class TestSupplementalAlgorithmDriver
+{
+public:
+  TestSupplementalAlgorithmDriver(
+    const stk::mesh::BulkData& bulk)
+    : bulk_(bulk)
+  {}
+
+  void nodal_execute()
+  {
+    const stk::mesh::MetaData& meta = bulk_.mesh_meta_data();
+
+    const stk::mesh::Selector selector = meta.locally_owned_part();
+
+    const auto& buckets = bulk_.get_buckets(stk::topology::NODE_RANK,
+                                            selector);
+
+    kokkos_thread_team_bucket_loop(buckets, [&](stk::mesh::Entity node) {
+
+      double lhs_value = 0.0;
```

Comment:
@marchdf Should the `lhs_value` and `rhs_value` be re-initialized within the `for` loop below? The SuppAlgs usually do a `+=` operation on the LHS and RHS entries.